src/solution.c

```diff
@@ -254,7 +262,30 @@ static void output_baseline(u8 num_sdiffs, const sdiff_t *sdiffs,
     break;
   }
 
-  solution_send_baseline(t, num_used, b, position_solution.pos_ecef, flags, hdop, diff_time, base_id);
+  /* Calculate the DOPs from the sids used in the baseline solution. */
+  gnss_solution soln;
+  dops_t dops;
+  navigation_measurement_t intersection_nm[MAX_CHANNELS];
+  /* Reusing nav_meas for the output. Should be safe as output will be same
+     length or shorter than the input. */
+  s32 intersection_size = intersection(num_used, sizeof(gnss_signal_t), used_sids, NULL,
+                          num_meas, sizeof(navigation_measurement_t), nav_meas, intersection_nm,
+                          cmp_sid_nm);
+  if (intersection_size > 0) {
+    /* The DOPs are always valid even if calc_PVT fails. */
+    calc_PVT(intersection_size, intersection_nm, TRUE, &soln, &dops);
+    log_debug("RTK DOPs: H %.2f V %.2f P %.2f T %.2f G %.2f",
+              dops.hdop, dops.vdop, dops.pdop, dops.tdop, dops.gdop);
+
+    /* Ensure PDOP is not too high. */
+    if (dops.pdop <= 20.0f) {
+      solution_send_baseline(t, num_used, b, position_solution.pos_ecef, flags, dops.hdop, diff_time, base_id);
+    } else {
+      log_error("output_baseline: RTK PDOP %.2f > 20", dops.pdop);
```

Comment:
Sure, what error level is the SPP errors logged at? Since that also triggers on high DOP.